src/main/java/it/smartcommunitylab/aac/files/service/FileInfoService.java

```diff
@@ -0,0 +1,161 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+import it.smartcommunitylab.aac.common.AlreadyRegisteredException;
+import it.smartcommunitylab.aac.common.NoSuchResourceException;
+import it.smartcommunitylab.aac.common.NoSuchUserException;
+import it.smartcommunitylab.aac.dto.FileInfoDTO;
+import it.smartcommunitylab.aac.files.persistence.FileInfoEntity;
+import it.smartcommunitylab.aac.files.persistence.FileInfoRepository;
+import java.util.ArrayList;
+
+@Service
+@Transactional
+public class FileInfoService {
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+
+	@Autowired
+	private FileInfoRepository fileInfoRepository;
+
+	public FileInfoService(FileInfoRepository fileInfoRepository) {
+		Assert.notNull(fileInfoRepository, "fileinfo repository is mandatory");
+		this.fileInfoRepository = fileInfoRepository;
+		logger.debug("fileinfo service initialized");
+	}
+
+	public FileInfoEntity createFileInfo(String realm) {
+		FileInfoEntity f = new FileInfoEntity();
+		f.setId(generateUuid());
+		f.setRealm(realm);
+		return f;
+	}
+
+	public FileInfoEntity addFileInfo(String id, String realm, String fileName, String mimeType, long size)
+			throws AlreadyRegisteredException {
+		logger.debug("Add file for realm {}", String.valueOf(realm));
+		FileInfoEntity exist = fileInfoRepository.findOne(id);
+		if (exist != null) {
+			throw new AlreadyRegisteredException();
+		}
+		// create file.
+		if (id == null || id.isBlank() || id.isEmpty()) {
+			id = generateUuid();
+		}
+		FileInfoEntity f = new FileInfoEntity();
+		f.setId(id);
+		f.setRealm(realm);
+		f.setName(fileName);
+		f.setMimeType(mimeType);
+		f.setSize(size);
+		f = fileInfoRepository.save(f);
+		return f;
+	}
+
+	@Transactional(readOnly = true)
+	public FileInfoEntity getFile(String id) throws NoSuchUserException {
+		logger.debug("Read file id {}", String.valueOf(id));
+		FileInfoEntity f = fileInfoRepository.findOne(id);
+		if (f == null) {
+			throw new NoSuchUserException("no file with id " + id);
+		}
+		return f;
+	}
+
+	@Transactional(readOnly = true)
+	public long countFiles(@NotNull String realm) {
+		return fileInfoRepository.countByRealm(realm);
+	}
+
+	@Transactional(readOnly = true)
+	public List<FileInfoDTO> listFiles(@NotNull String realm) {
+		List<FileInfoDTO> listfilesDTO = new ArrayList<>();
+		for (FileInfoEntity fileInfoEntity: fileInfoRepository.findByRealm(realm)) {
+			listfilesDTO.add(getFileInfoDTO(fileInfoEntity));
+		}		
+		return listfilesDTO;
+	}
+
+	public List<FileInfoEntity> readAllFileInfo() {
+		return fileInfoRepository.findAll();
+	}
+
+	public FileInfoDTO readFileInfo(String realm, String id) throws FileNotFoundException {
+		FileInfoEntity file = fileInfoRepository.findByRealmAndId(realm, id);
+		if (file != null) {
+			return getFileInfoDTO(file);
+		}
+		throw new FileNotFoundException("file not found");
```

Comment:
replace FileNotFoundException with a dedicated NoSuchFileException. The first one indicates a missing file *blob* on filesystem, while here we are looking at metadata in a db